channel-email/src/test/java/com/synopsys/integration/alert/channel/email/action/EmailGlobalTestActionTest.java

```diff
@@ -91,12 +103,58 @@ public void testConfigMissingDestinationTest() {
     public void testConfigInvalidDestinationTest() {
         AuthorizationManager authorizationManager = createAuthorizationManager(AuthenticationTestUtils.FULL_PERMISSIONS);
         EmailGlobalConfigurationValidator validator = new EmailGlobalConfigurationValidator();
-        EmailGlobalTestAction emailGlobalTestAction = new EmailGlobalTestAction(authorizationManager, validator, null, null);
+        EmailGlobalTestAction emailGlobalTestAction = new EmailGlobalTestAction(authorizationManager, validator, null, null, null);
 
         ConfigurationTestResult testResult = emailGlobalTestAction.testConfigModelContent("not a valid email address", new EmailGlobalConfigModel());
         assertFalse(testResult.isSuccess());
     }
 
+    @Test
+    public void testSmtpPasswordMissingTest() throws AlertException {
+        AuthorizationManager authorizationManager = createAuthorizationManager(255);
+        EmailGlobalConfigurationValidator validator = new EmailGlobalConfigurationValidator();
+        EmailChannelMessagingService emailChannelMessagingService = Mockito.mock(EmailChannelMessagingService.class);
+        Mockito.when(emailChannelMessagingService.sendMessage(Mockito.any(), Mockito.any())).thenReturn(new MessageResult("PASS"));
+
+        JavamailPropertiesFactory javamailPropertiesFactory = new JavamailPropertiesFactory();
+        EmailGlobalTestAction emailGlobalTestAction = new EmailGlobalTestAction(authorizationManager, validator, emailChannelMessagingService, javamailPropertiesFactory, configurationAccessor);
+
+        EmailGlobalConfigModel emailGlobalConfigModel = new EmailGlobalConfigModel();
+        emailGlobalConfigModel.setIsSmtpPasswordSet(true);
+        EmailGlobalConfigModel configModelWithPassword = new EmailGlobalConfigModel();
+        configModelWithPassword.setIsSmtpPasswordSet(true);
+        configModelWithPassword.setSmtpPassword("password");
+        Mockito.when(configurationAccessor.getConfigurationByName(AlertRestConstants.DEFAULT_CONFIGURATION_NAME)).thenReturn(Optional.of(configModelWithPassword));
+
+        ConfigurationTestResult testResult = emailGlobalTestAction.testConfigModelContent("noreply@synopsys.com", emailGlobalConfigModel);
+        assertTrue(testResult.isSuccess(), "Expected the message result to not have errors");
+    }
+
+    @Test
+    @Tags(value = {
+        @Tag(TestTags.DEFAULT_INTEGRATION),
+        @Tag(TestTags.CUSTOM_EXTERNAL_CONNECTION)
+    })
+    public void testSmtpPasswordMissingTestIT() {
+        AuthorizationManager authorizationManager = createAuthorizationManager(AuthenticationTestUtils.FULL_PERMISSIONS);
+        EmailGlobalConfigurationValidator validator = new EmailGlobalConfigurationValidator();
+        TestProperties testProperties = new TestProperties();
+        String emailAddress = testProperties.getProperty(TestPropertyKey.TEST_EMAIL_RECIPIENT);
+
+        assumeTrue(testProperties.getOptionalProperty(TestPropertyKey.TEST_EMAIL_SMTP_PASSWORD).isPresent());
```

Comment:
Adding an assumeTrue check for the SMTP password and skip if it is not set.  Without it we will not be correctly test the problem when attempting to test a configuration with an obfuscated password